src/common/state/redis.cc

```diff
@@ -672,7 +673,8 @@ void redis_object_table_lookup_callback(redisAsyncContext *c,
     for (size_t j = 0; j < reply->elements; ++j) {
       CHECK(reply->element[j]->type == REDIS_REPLY_STRING);
       DBClientID manager_id;
-      memcpy(manager_id.id, reply->element[j]->str, sizeof(manager_id.id));
+      memcpy(manager_id.mutable_data(), reply->element[j]->str,
```

Comment:
We have a decent number of `memcpy`'s in the codebase. Maybe this functionality should be part of the `ObjectID` class.

Reply:
This code will go away as soon as the new GCS client code will be merged, and the rest should be modernized piece by piece.